web_ui/server.py

```
@@ -1108,6 +1110,417 @@ def _debug_process_snapshot(session_id: str | None = None) -> dict[str, object]:
+        "book_series_index": _stringify_preview_value(meta_data.get("book_series_index")),
+        "platform": _stringify_preview_value(meta_data.get("platform")),
+        "game_version": _stringify_preview_value(meta_data.get("game_version")),
+        "game_subcategory": _stringify_preview_value(meta_data.get("game_subcategory")),
+        "game_region": _stringify_preview_value(meta_data.get("game_region")),
+        "game_system": _stringify_preview_value(meta_data.get("game_system")),
+        "developer": _stringify_preview_value(meta_data.get("developer")),
+        "awaiting_input": False,
+    }
+
+
+def _find_execution_preview(session_id: str) -> ExecutionPreview | None:
```

Ruff (0.15.21)

[error] Undefined name `ExecutionPreview`

(F821)